### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by Tshobohwa**
# In this milestone I implemented the following features:

1.  Fixed the n+1 issue while fetching all the posts for a user with their comments,
- query before fixing the n+1 issue:
queriesn1
- Query after fixing the n+1 issue:
queriesn2

2.  Used Capybara to create tests for:
- User index page.
- user show page.
- post index page.
- post show page .

### CODE REVIEW

File: spec/integrations/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  it 'It should display the count of likes of each post' do
+    page.has_content?(@post.likes_counter)
+  end
+
+  it 'It should show the pagination' do
+    page.has_content?('Next')
+  end
+end
```
Comment (zieeco) [inline spec/integrations/post_index_spec.rb:53]: **[REQUIRED]**

- Hey, let's also add the test that redirects to the `post's show page` when you click on a post.

File: spec/integrations/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    page.has_content?(@post.likes_counter)
+  end
+
+  it 'It should display the post body.' do
+    page.has_content?(@post.text)
+  end
+  it 'It should display the author of each commentor.' do
+    page.has_content?(@comment.author.name)
+  end
+end
```
Comment (zieeco) [inline spec/integrations/post_show_spec.rb:31]: ## comment_spec.rb

**[optioonal[**

- Your test here is failing, please, try fixing it. your other test for the integration passed as expected. doulbe-check this test file to fix the failing test.

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,





Great job so far! 👏 
You've done well, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Fix n+1 problem.✔️
- Integration test passed. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Check the comments under the review._

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Zuhaib042 — APPROVED** [review]
Hi @Tshobohwa , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Well implemented suggested changes
✅ Integration tests are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**